src/lib.rs

```diff
@@ -413,14 +413,14 @@ where
     ///
     /// **Note:** For performance reasons, the `sleep` flag is not being
     /// checked here. Make sure not to call this method while in sleep mode.
-    fn write_byte(&mut self, byte: u8) -> RnResult<()> {
-        block!(self.serial.write(byte)).map_err(|_| Error::SerialWrite)
+    fn write_byte(&mut self, byte: u8) -> RnResult<(), E> {
+        block!(self.serial.write(byte)).map_err(|e| Error::SerialWrite(e))
```

Comment:
As suggested by clippy, you can replace the closure:

```suggestion
        block!(self.serial.write(byte)).map_err(Error::SerialWrite)
```

Reply:
There are a bunch of other clippy issues:
<details>

```
warning: name `RN2483` contains a capitalized acronym
   --> src/lib.rs:236:5
    |
236 |     RN2483,
    |     ^^^^^^ help: consider making the acronym lowercase, except the initial letter: `Rn2483`
    |
    = note: `#[warn(clippy::upper_case_acronyms)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms

warning: name `RN2903` contains a capitalized acronym
   --> src/lib.rs:237:5
    |
237 |     RN2903,
    |     ^^^^^^ help: consider making the acronym lowercase, except the initial letter: `Rn2903`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms

warning: manual `RangeInclusive::contains` implementation
  --> src/utils.rs:58:8
   |
58 |     if port >= 1 && port <= 223 {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `(1..=223).contains(&port)`
   |
   = note: `#[warn(clippy::manual_range_contains)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains

warning: manual `!RangeInclusive::contains` implementation
   --> src/lib.rs:641:12
    |
641 |         if addr < 0x300 || addr > 0x3ff {
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `!(0x300..=0x3ff).contains(&addr)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains

warning: manual `!RangeInclusive::contains` implementation
   --> src/lib.rs:665:12
    |
665 |         if addr < 0x300 || addr > 0x3ff {
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `!(0x300..=0x3ff).contains(&addr)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains

warning: 5 warnings emitted
```
</details>

It would be easier to just fix these and fail CI if there are any clippy issues.